arch/lib/include/asm/barrier.h

```diff
@@ -1,8 +1,105 @@
-#include <asm-generic/barrier.h>
+#ifndef __ASM_BARRIER_H
```

Comment:
(to remind private message to @jyizheng )

1) I would like to see whether those include/asm code taken from arch/arm can replace with asm-generic headers or not.
2) if there is a certain reason to use another arch code (e.g., arm), then symbolic link or copying automatically during make would be nicer than stored in arch/lib/include.